beacon_node/network/src/router/processor.rs

```diff
@@ -246,64 +215,26 @@ impl<T: BeaconChainTypes> Processor<T> {
                 .root_at_slot(start_slot(remote.finalized_epoch))
                 .map(|root_opt| root_opt != Some(remote.finalized_root))?
         {
-            // The remotes finalized epoch is less than or greater than ours, but the block root is
-            // different to the one in our chain.
-            //
-            // Therefore, the node is on a different chain and we should not communicate with them.
-            debug!(
-                self.log, "Handshake Failure";
-                "peer" => peer_id.to_string(),
-                "reason" => "different finalized chain"
-            );
+            // The remote's finalized epoch is less than or equal to ours, but the block root is
+            // different to the one in our chain. Therefore, the node is on a different chain and we
+            // should not communicate with them.
+            Some("Different finalized chain")
+        } else {
+            None
+        };
+
+        if let Some(irrelevant_reason) = irrelevant_reason {
+            debug!(self.log, "Handshake Failure"; "peer" => %peer_id, "reason" => irrelevant_reason);
             self.network
                 .goodbye_peer(peer_id, GoodbyeReason::IrrelevantNetwork);
-        } else if remote.finalized_epoch < local.finalized_epoch {
-            // The node has a lower finalized epoch, their chain is not useful to us. There are two
-            // cases where a node can have a lower finalized epoch:
-            //
-            // ## The node is on the same chain
-            //
-            // If a node is on the same chain but has a lower finalized epoch, their head must be
-            // lower than ours. Therefore, we have nothing to request from them.
-            //
-            // ## The node is on a fork
-            //
-            // If a node is on a fork that has a lower finalized epoch, switching to that fork would
-            // cause us to revert a finalized block. This is not permitted, therefore we have no
-            // interest in their blocks.
-            debug!(
-                self.log,
-                "NaivePeer";
-                "peer" => peer_id.to_string(),
-                "reason" => "lower finalized epoch"
-            );
-        } else if self
-            .chain
-            .store
-            .item_exists::<SignedBeaconBlock<T::EthSpec>>(&remote.head_root)?
-        {
-            debug!(
-                self.log, "Peer with known chain found";
```

Comment:
I think these logs in some form are still useful. 

I know I still use them to diagnose sync issues. Happy to remove if you think they are irrelevant.

Reply:
I find them a bit misleading since they get printed with every status even if the peer is not newly connected and because they didn't match 100% to sync's view . We still have the debug logs of  "Received Status Request" with the Status info and the debug log if the peer's sync status changes (which now will be printed if passing from unknown to something else = first time the peer is connected) 
Do you think we are missing something if left only with those?